### PR TITLE
CI minimum mpl version everywhere

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [ ] closes #50202  (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: ci/deps/actions-38-downstream_compat.yaml [hunk 1]
```diff
@@ -34,7 +34,7 @@ dependencies:
   - gcsfs
   - jinja2
   - lxml
-  - matplotlib
+  - matplotlib>=3.6.1
```
Comment (phofl) [inline ci/deps/actions-38-downstream_compat.yaml:37]: I think we should do this for every dependency. cc @mroeschke

Comment (phofl) [reply ci/deps/actions-38-downstream_compat.yaml:37]: Yeah sure.

**Review by phofl — APPROVED** [review]

File: ci/deps/actions-38-downstream_compat.yaml (hunk 1 above)
Comment (mroeschke) [reply ci/deps/actions-38-downstream_compat.yaml:37]: Agreed, and to update our min version script to account for these dependency files

**Comment by phofl** [comment]
@MarcoGorelli could you look into 1.5.x? We get failures there as well but will need a different min version probably

**Comment by MarcoGorelli** [comment]
🤔 don't know, the minimum version is set to 3.3.2 in that branch, and that's exactly what CI is pulling in. will investigate

**Comment by MarcoGorelli** [comment]
@meeseeks-dev backport to 1.5.x

**Comment by MarcoGorelli** [comment]
@meeseeksdev backport to 1.5.x

**Pull request merged**